src/Core/Prefab/computeBufferTileGeometry.ts

```diff
@@ -0,0 +1,338 @@
+import type { TileBuilder, TileBuilderParams } from 'Core/Prefab/TileBuilder';
+import * as THREE from 'three';
+
+export function getBufferIndexSize(segments: number, noSkirt: boolean): number {
+    const triangles = (segments) * (segments) * 2
+        + (noSkirt ? 0 : 4 * segments * 2);
+    return triangles * 3;
+}
+
+type Option<T> = T | undefined;
+
+type IndexArray = Option<Uint8Array | Uint16Array | Uint32Array>;
+
+export type Buffers = {
+    index: IndexArray,
+    position: Float32Array,
+    normal: Float32Array,
+    uvs: [Option<Float32Array>, Option<Float32Array>],
+};
+
+type TmpBuffers = Buffers & {
+    skirt: IndexArray,
+};
+
+function pickUintArraySize(
+    highestValue: number,
+): Uint8ArrayConstructor | Uint16ArrayConstructor | Uint32ArrayConstructor {
+    let picked = null;
+
+    if (highestValue < 2 ** 8) {
+        picked = Uint8Array;
+    } else if (highestValue < 2 ** 16) {
+        picked = Uint16Array;
+    } else if (highestValue < 2 ** 32) {
+        picked = Uint32Array;
+    } else {
+        throw new Error('Value is too high');
+    }
+
+    return picked;
+}
+
+function allocateIndexBuffer(
+    nVertex: number,
+    nSeg: number,
+    params: TileBuilderParams,
+): Option<{ index: IndexArray, skirt: IndexArray }> {
+    if (!params.buildIndexAndUv_0) {
+        return undefined;
+    }
+
+    const indexBufferSize = getBufferIndexSize(nSeg, params.disableSkirt);
+    const indexConstructor = pickUintArraySize(nVertex);
+
+    const tileLen = indexBufferSize;
+    const skirtLen = 4 * nSeg;
```

Comment:
The 4 here is the number of vertices per segment of the edge since we're creating squares. The rest of the generation wouldn't work with anything different (either failing out of bounds checks or leaving trailing uninitialized memory). I also struggle to see a use case for doing something other than squares to be quite honest, do you have any in mind?

Reply:
Oh yes sorry, I thought it was the actual lengh of the skirt in the view, I must have been tired at this point of the review :sweat_smile: